src/common/rabbitmq/rabbitmq.nft.handler.service.ts

```diff
@@ -0,0 +1,28 @@
+import { Injectable } from '@nestjs/common';
+import { NftService } from 'src/endpoints/nfts/nft.service';
+import { ProcessNftSettings } from 'src/endpoints/process-nfts/entities/process.nft.settings';
+import { NftWorkerService } from 'src/queue.worker/nft.worker/nft.worker.service';
+import { NftCreateEvent } from './entities/nft/nft-create.event';
+
+@Injectable()
+export class RabbitMqNftHandlerService {
+
+  constructor(
+    private readonly nftWorkerService: NftWorkerService,
+    private readonly nftService: NftService,
+  ) { }
+
+  public async handleNftCreateEvent(event: NftCreateEvent): Promise<void> {
+    const identifier = event.getTopics()?.identifier;
+
+    const nft = await this.nftService.getSingleNft(identifier);
+    if (!nft) {
+      return;
+    }
+
+    const needsProcessing = await this.nftWorkerService.needsProcessing(nft, new ProcessNftSettings());
```

Comment:
Also, the nft processing should be surrounded with try catch and log if an error occurs related to processing 'NFTProcess: Unexpected error when processing NFT with identifier'